src/app/topbar/topbar-container.component.ts

```diff
@@ -1,24 +1,65 @@
 import { Component, HostBinding } from '@angular/core';
+import { JwtHelper } from 'angular2-jwt';
+import { AuthService } from '../services/auth.service';
 
 @Component({
   selector: 'app-topbar-container',
   template: `
     <nav class="nav-left">
-      <a *ngIf="apiTokenEnabled" class="btn btn-default">Set API Token</a>
-      <a *ngIf="loginEnabled" class="login-link">Login</a>
+      <input type="text"
+             class="form-control nav-control"
+             [value]="token"
+             (input)="validateToken($event.target.value)">
+      <button type="button"
+              *ngIf="apiTokenEnabled"
+              class="btn btn-default btn-navbar"
+              [disabled]="isSetTokenDisabled"
+              (click)="authService.setToken(token)">
+        Set API Token
+      </button>
+      <button *ngIf="loginEnabled" class="btn btn-navbar">Login</button>
     </nav>
     <nav class="nav-right">
       <a href="#" class="btn btn-secondary">curl</a>
     </nav>
   `
 })
 export class TopbarContainerComponent {
+  private error: string = 'No Token';
+  private jwtHelper: JwtHelper = new JwtHelper();
+  private token: string = '';
 
-  constructor() { }
+  constructor(private authService: AuthService) { }
 
   @HostBinding('class')
   private classNames: string = 'navbar';
 
-  private apiTokenEnabled: boolean = false;
+  private apiTokenEnabled: boolean = true;
```

Comment:
I believe the `private` here and below is causing an issue when building (see #7) because we use them in the template. Sad. Could be an opportunity to fix.